google/cloud/aiplatform/datasets/dataset.py

```diff
@@ -52,23 +64,89 @@ def __init__(
                 "456" when project and location are initialized or passed.
-            project (str):
+            project: (str) = None
                 Optional project to retrieve dataset from. If not set, project
                 set in aiplatform.init will be used.
-            location (str):
+            location: (str) = None
                 Optional location to retrieve dataset from. If not set, location
                 set in aiplatform.init will be used.
-            credentials: Optional[auth_credentials.Credentials]=None,
+            credentials: Optional[auth_credentials.Credentials] = None,
                 Custom credentials to use to upload this model. Overrides
                 credentials set in aiplatform.init.
+
+        Raises:
+            ValueError if the dataset type of the retrieved dataset resource is
+            not supported by the 'Dataset' class.
         """
 
         super().__init__(project=project, location=location, credentials=credentials)
         self._gca_resource = self._get_gca_resource(resource_name=dataset_name)
+        self._validate_resource_metadata_schema_uri()
+
+    @property
+    def resource_metadata_schema_uri(self) -> str:
+        """The metadata schema uri of this dataset resource."""
+        return self._gca_resource.metadata_schema_uri
+
+    def _validate_resource_metadata_schema_uri(self) -> bool:
+        if self._support_metadata_schema_uris and (
+            self.resource_metadata_schema_uri not in self._support_metadata_schema_uris
+        ):
+            raise ValueError(
+                f"{self.__class__.__name__} class can not be used to retrieve "
+                f"dataset resource {self.resource_name}, check the dataset type"
+            )
+        return True
+
+    @property
+    def metadata_schema_uri(self) -> str:
+        """The metadata schema uri for creating dataset."""
+        return self._metadata_schema_uri
+
+    @metadata_schema_uri.setter
+    def metadata_schema_uri(self, metadata_schema_uri):
+        if self._support_metadata_schema_uris and (
+            metadata_schema_uri not in self._support_metadata_schema_uris
+        ):
+            raise ValueError(
+                f"{self.__class__.__name__} class does not support "
+                f"{metadata_schema_uri}"
+            )
+        self._metadata_schema_uri = metadata_schema_uri
+
+    @property
+    def import_schema_uri(self) -> str:
+        """The import schema uri for importing data."""
+        return self._import_schema_uri
+
+    @import_schema_uri.setter
+    def import_schema_uri(self, import_schema_uri):
+        if not self._support_import_schema_classes:
+            raise ValueError(
+                f"{self.__class__.__name__} class does not support importing data"
+            )
+
+        for import_schema_class in self._support_import_schema_classes:
+            if (
+                self.metadata_schema_uri
```

Comment:
This creates a temporal dependency such that the user has to first set `self.metadata_schema_uri` before setting `self.import_schema_uri`, which can introduce bugs, since both are setters.

Again, I would prefer not to have property setters if possible.
We could perhaps provide getters if that's something users care about. I'd lean towards hiding as much as possible.